src/igl/metal/RenderCommandEncoder.mm

```diff
@@ -384,7 +384,18 @@
                 baseInstance:baseInstance];
 #if IGL_PLATFORM_IOS
   } else {
-    [encoder_ drawPrimitives:metalPrimitive_ vertexStart:firstVertex vertexCount:vertexCount];
+    if (baseInstance) {
+      IGL_DEBUG_ASSERT_NOT_IMPLEMENTED();
+    } else {
+      if (instanceCount != 1) {
+        [encoder_ drawPrimitives:metalPrimitive_
+                     vertexStart:firstVertex
+                     vertexCount:vertexCount
+                   instanceCount:instanceCount];
```

Comment:
`instanceCount` is not supported everywhere https://developer.apple.com/documentation/metal/mtlrendercommandencoder/drawindexedprimitives(type:indexcount:indextype:indexbuffer:indexbufferoffset:instancecount:)

It needs an extra check.